integration/helper/namespace.go

```diff
@@ -84,6 +49,7 @@ func tempCreateHelper(t *testing.T, client client.Client, namespaceObject corev1
 
 	t.Cleanup(func() {
 		namespaceDeleting := namespaceObject.Name
+		_ = client.Get(ctx, k8sclient.ObjectKey{Name: namespaceObject.Name}, &namespaceObject)
```

Comment:
Thought it could not be worth handling errors in the cleanup function. But could be useful doing to print a message if it does fail to clean up. I'll add something!